internal/authorization/serviceaccount_authenticator_test.go

```diff
@@ -105,12 +156,29 @@ func TestBasicAuthenticatorRoundTrip(t *testing.T) {
 func testServerForServiceAccount(t *testing.T) *httptest.Server {
 	t.Helper()
 	return testServer(t, func(w http.ResponseWriter, r *http.Request) {
+		contentType := r.Header.Get("Content-Type")
 		switch {
-		case r.Method == http.MethodPost && r.RequestURI == basicAuthEndpoint:
+		case r.Method == http.MethodPost && r.RequestURI == serviceAccountAuthEndpoint && contentType == "application/x-www-form-urlencoded":
```

Comment:
This content-type is used both in tests and code. We could create a constant on this, what do you think?